src/vs/workbench/contrib/terminal/browser/terminalSpeechToText.ts

```diff
@@ -0,0 +1,203 @@
+	) {
+		super();
+		this._register(this._terminalService.onDidChangeActiveInstance(() => this.stop()));
+		this._register(this._terminalService.onDidDisposeInstance(() => this.stop()));
+	}
+
+	start(): void {
+		this.stop();
+		let voiceTimeout = this.configurationService.getValue<number>(AccessibilityVoiceSettingId.SpeechTimeout);
+		if (!isNumber(voiceTimeout) || voiceTimeout < 0) {
+			voiceTimeout = SpeechTimeoutDefault;
+		}
+		this._acceptTranscriptionScheduler = this._disposables.add(new RunOnceScheduler(() => {
+			this._terminalService.activeInstance?.sendText(this._input, false);
+			this.stop();
+		}, voiceTimeout));
+		this._cancellationTokenSource = new CancellationTokenSource();
+		const session = this._disposables.add(this._speechService.createSpeechToTextSession(this._cancellationTokenSource!.token));
+
+		this._disposables.add(session.onDidChange((e) => {
+			if (this._cancellationTokenSource?.token.isCancellationRequested) {
+				return;
+			}
+			switch (e.status) {
+				case SpeechToTextStatus.Started:
+					// TODO: play start audio cue
+					if (!this._decoration) {
+						this._createDecoration();
+					}
+					break;
+				case SpeechToTextStatus.Recognizing: {
+					this._updateInput(e);
+					this._renderGhostText(e);
+					if (voiceTimeout > 0) {
+						this._acceptTranscriptionScheduler!.cancel();
+					}
+					break;
+				}
+				case SpeechToTextStatus.Recognized:
+					this._updateInput(e);
+					if (voiceTimeout > 0) {
+						this._acceptTranscriptionScheduler!.schedule();
+					}
+					break;
+				case SpeechToTextStatus.Stopped:
+					// TODO: play stop audio cue
+					this.stop();
+					break;
+			}
+		}));
+	}
+	stop(send?: boolean): void {
+		this._setInactive();
+		if (send) {
+			this._acceptTranscriptionScheduler!.cancel();
+			this._terminalService.activeInstance?.sendText(this._input, false);
+		}
+		this._marker?.dispose();
+		this._ghostTextMarker?.dispose();
+		this._ghostText?.dispose();
+		this._ghostText = undefined;
+		this._decoration?.dispose();
+		this._decoration = undefined;
+		this._cancellationTokenSource?.cancel();
+		this._disposables.clear();
+		this._input = '';
+	}
+
+	private _updateInput(e: ISpeechToTextEvent): void {
+		if (e.text) {
+			let input = e.text.replaceAll(/[.,?;!]/g, '');
+			const symbolMap: { [key: string]: string } = {
+				'Ampersand': '&',
+				'ampersand': '&',
+				'Dollar': '$',
+				'dollar': '$',
+				'Percent': '%',
+				'percent': '%',
+				'Asterisk': '*',
+				'asterisk': '*',
+				'Plus': '+',
+				'plus': '+',
+				'Equals': '=',
+				'equals': '=',
+				'Exclamation': '!',
+				'exclamation': '!',
+				'Slash': '/',
+				'slash': '/',
+				'Backslash': '\\',
+				'backslash': '\\',
+				'Dot': '.',
+				'dot': '.',
+				'Period': '.',
+				'period': '.',
+				'Quote': '\'',
+				'quote': '\'',
+				'double quote': '"',
+				'Double quote': '"',
+			};
+
+			for (const symbol in symbolMap) {
```

Comment:
You likely want to replace whole words. You can use word boundaries `\b` in the regular expression 

Also:

- probably worth extracting the object map to a static constant instead of reconstructing it every time 
- Avoid `for in` loops as they can iterate through unexpected properties. Instead use a `Map` or `for of` with `Object.entries(obj)`